### PR TITLE
users: Update information on CSV import

### PR DESCRIPTION
**Pull request by greezybacon**
This patch adds a feature to the import process, so that if a user already exists (identified by its email address), then the name and other custom data are updated with the information in the CSV record being imported.

### Outstanding
* [x] Touch DynamicFormEntry::updated when custom data is updated

### CODE REVIEW

File: include/class.user.php [hunk 1]
```diff
@@ -196,6 +196,20 @@ static function fromVars($vars) {
                 return null;
             }
         }
+        elseif ($update) {
+            foreach ($user->getDynamicData() as $entry) {
```
Comment (greezybacon) [inline include/class.user.php]: Use `$this->updateInfo()`

**Comment by Remo** [comment]
Shouldn't this code result in a user with the name `bbb`?

```php
$data = [
    'name' => 'aaa',
    'email' => 'info@aaa.ch',
    'phone' => 123,
];
$user = User::fromVars($data, true);

$data = [
    'name' => 'bbb',
    'email' => 'info@aaa.ch',
    'phone' => 456,
];
$user = User::fromVars($data, true);
```

**Comment by ntozier** [comment]
email is a unique field, so no.

**Comment by Remo** [comment]
Yes, it is, but this line added an implicit update, didn't it?
https://github.com/osTicket/osTicket-1.8/pull/1993/files#diff-051ae6fc9b5c9f4276374b24cb7bd5b6R166

I would expect that the user gets updated after merging this change..

**Comment by greezybacon** [comment]
@Remo we've had significant troubles with the custom data system including strange issues like this. We made some very significant changes in the `develop-next` branch to remove most of the "magic" employed to make it work in this version. Could you try your code against the code in the `develop-next` branch to see if we've already fixed this and should just consider back-porting the changes?

**Comment by Remo** [comment]
Looks good, will run a more sophisticated test on Thurday

**Comment by Remo** [comment]
@greezybacon the implicit update didn't work with `develop-next`, but writing multiple fields at once worked fine

**Pull request merged**